stable/kibana/templates/configmap-dashboardimport.yaml

```diff
@@ -21,20 +21,25 @@ data:
 
     /usr/local/bin/kibana-docker $@ &
 
-    echo "Waiting up to 60 seconds for Kibana to get in green overall state..."
+    echo "Waiting up to {{ .Values.dashboardImport.timeout }} seconds for Kibana to get in green overall state..."
 
-    for i in {1..60}; do
+    for i in {1..{{ .Values.dashboardImport.timeout }}}; do
       curl -s localhost:5601/api/status | python -c 'import sys, json; print json.load(sys.stdin)["status"]["overall"]["state"]' 2> /dev/null | grep green > /dev/null && break || sleep 1
     done
 
     for DASHBOARD_FILE in *; do
       echo -e "Importing ${DASHBOARD_FILE} dashboard..."
 
-      curl -v {{ if .Values.dashboardImport.xpackauth.enabled }}--user {{ .Values.dashboardImport.xpackauth.username }}:{{ .Values.dashboardImport.xpackauth.password }}{{ end }} -s --connect-timeout 60 --max-time 60 -XPOST localhost:5601/api/kibana/dashboards/import?force=true -H 'kbn-xsrf:true' -H 'Content-type:application/json' -d @./${DASHBOARD_FILE}
+      if [ ! "$(python -c 'import sys, json; print json.load(sys.stdin)' < ${DASHBOARD_FILE} &> /dev/null)" ]; then
+        echo "${DASHBOARD_FILE} is no valid json. Trying to import from url..."
```

Comment:
But is it valid URL? It may be just a typo in the dashboard.
It may be better to split this into two variables so that what is happening is crystal clear.

Reply:
If it's no valid url it fails...

Reply:
Just saying it would be much cleaner by having two lists, and not mixing JSON with URLs. Not even speaking about faster debugging ("wait, it failed because it's not a valid url, but my dashboard is not an url...")